core/application/commands.go

```diff
@@ -26,17 +26,25 @@ func (a *app) CreateMicroVM(ctx context.Context, mvm *models.MicroVM) (*models.M
 			return nil, fmt.Errorf("generating random name for microvm: %w", err)
 		}
 
-		vmid, err := models.NewVMID(name, defaults.MicroVMNamespace)
+		vmid, err := models.NewVMID(name, defaults.MicroVMNamespace, "")
 		if err != nil {
 			return nil, fmt.Errorf("creating vmid: %w", err)
 		}
 
 		mvm.ID = *vmid
 	}
 
+	uuid, err := a.ports.IdentifierService.GenerateRandom()
```

Comment:
The infrastructure implementation that we use for the identifier service doesn't generate a UUID but instead a ULID. It does say a ULID is compatible with a UUID but its not formatted as a UUID with this call.

Maybe we need to refer to this as a uid instead of uuid to indicate that its not a UUID? Or something similar.

Reply:
actually, it's even more confusing now. Turned out i used `a.ports.IdentifierService.GenerateRandom()` here and in the grpc server `uuid.New().String()`  which one should we keep?

Reply:
i'll use ulid to me more consistent with the rest of the system.

Reply:
Yeah ulid from the identifier service is the best.